hw/xbox/mcpx_apu.c

```diff
@@ -24,8 +25,13 @@
 #include "hw/pci/pci.h"
 #include "cpu.h"
 #include "hw/xbox/dsp/dsp.h"
+#include "audio/audio.h"
+#include "qemu/fifo8.h"
 #include <math.h>
 
+#define USE_APU_THREAD 1
```

Comment:
Why is this an option?

We should do threading, or we shouldn't do it - I don't think there's a benefit of having it be an option; much less a compile-time option.

Giving the option just increases complexity and makes it harder to trace problems (if someone decides to mess with this).

Reply:
This comment should be on Audio PR#2 as this branch is based on it, but I'll respond here.

This is a holdover from my development so I could revert back quickly to test if there was an issue with the independent thread. I left it in so others could also help test the changes, but the threading approach is the way to go and it's very stable after my local testing over the past couple months. I'm going to clean up Audio PR#2 and remove timer model altogether.